wave_kinematics/include/wave/kinematics/numerical_test_functors.hpp

```diff
@@ -0,0 +1,78 @@
+#ifndef __WAVE_NUMERICAL_TEST_FUNCTORS_HPP__
+#define __WAVE_NUMERICAL_TEST_FUNCTORS_HPP__
+
+#include <iostream>
+#include <functional>
+
+#include "wave/kinematics/rotation.hpp"
+#include "wave/utils/utils.hpp"
+#include <kindr/Core>
+
+namespace wave {
+
+// The Functors are used for computing finite difference Jacobians
+// of rotation expressions.
+
+class RotateAndJacobianJpointFunctor {
+public:
+  wave::Rotation R;
+  wave::Vec3 P;
+  RotateAndJacobianJpointFunctor(wave::Rotation input_rotation) {
+    this->R = input_rotation;
+  }
+
+  int operator()(wave::Vec3 input_point, wave::Vec3 &output_point) {
+    output_point = R.rotate(input_point);
+  }
+};
+
+
+class RotateAndJacobianJparamFunctor {
+public:
+  wave::Rotation R;
+  wave::Vec3 P;
+  RotateAndJacobianJparamFunctor(wave::Rotation input_rotation,
+                                 wave::Vec3 input_point) {
+    this->R = input_rotation;
+    this->P = input_point;
+  }
+
+  int operator()(wave::Vec3 input_point, wave::Vec3 &output_point) {
+    wave::Rotation Rp = this->R;
+    Rp.manifoldPlus(input_point);
+    output_point = Rp.rotate(P);
+  }
+};
+
+
+// Implement the numerical differentiation.  Based on the Eigen::NumericalDiff
+// Module.
```

Comment:
I'll be implementing more member functions after this PR, which are expressions that return Rotation Objects.  To find the numerical Jacobians of such expressions, we need to replace normal minus with manifoldMinus when computing (val2 - val1), thus the numerical_jacobian function will undergo quite a few more changes.  Are licensing issues still a concern, in that case?